src/components/MatchRuleList.tsx

```diff
@@ -0,0 +1,279 @@
+      ...matchRules,
+    ])
+  }
+
+  // To delete a rule when saving, we add it to the toDelete array
+  const deleteMatchRule = async (id: string) => {
+    const ruleIndex = matchRules.findIndex(rule => rule.id === id)
+    setToDelete([...toDelete, matchRules[ruleIndex]])
+  }
+
+  const saveMatchRules = async () => {
+    const updatedMatchRules: MatchRule[] = []
+    const errors: string[] = []
+
+    toDelete.forEach(rule => {
+      // Rules that have not been created yet are skipped, since they will be gone with the re-render anyways
+      if (rule.createdAt) {
+        return matchRuleApi.delete(rule).catch(err => errors.push(err.message))
+      }
+    })
+
+    // After saving, the list of rules to delete should be empty again.
+    setToDelete([])
+
+    const requests = matchRules.map((rule, index) => {
+      // If the rule is missing an account ID or a match, it is invalid, so we discard it
+      if (!rule.accountId || !rule.match) {
+        errors.push(
+          t('matchRules.invalid', {
+            match: rule.match || '',
+            account: rule.accountId || '',
+          })
+        )
+        return true
+      }
+
+      if (rule.createdAt) {
+        // createdAt is set by the backend, so this is an existing rule that needs to be updated
+        rule.priority = index
+
+        return matchRuleApi
+          .update(rule)
+          .then(updatedRule => {
+            updatedMatchRules.push(updatedRule)
+          })
+          .catch(err => errors.push(err.message))
+      } else {
+        // This is a new rule, it needs to be created
+        rule.priority = index
+        return matchRuleApi
+          .create(rule, budget)
+          .then(newRule => {
+            updatedMatchRules.push(newRule)
+          })
+          .catch(err => errors.push(err.message))
+      }
+    })
+
+    await Promise.all(requests)
+
+    if (errors.length > 0) {
+      setError(errors.join(', '))
+    }
+
+    setMatchRules(updatedMatchRules.sort((a, b) => a.priority - b.priority))
+    setNotification(t('changesSaved'))
+  }
+
+  // Delete matchRule when it is added to toDelete
+  useEffect(() => {
+    setMatchRules(matchRules.filter(matchRule => !toDelete.includes(matchRule)))
+  }, [toDelete])
+
+  useEffect(() => {
+    if (!isLoading) {
+      setIsLoading(true)
+    }
+
+    const promises = [
+      matchRuleApi.getAll(budget).then(setMatchRules),
+      accountApi.getAll(budget).then(setAccounts),
+    ]
+
+    Promise.all(promises)
+      .then(() => setError(''))
+      .catch(err => {
+        setError(err.message)
+      })
+      .then(() => setIsLoading(false))
+  }, [budget])
+
+  return (
+    <>
+      <div className="header mb-4">
+        <h1>{t('matchRules.matchRules')}</h1>
+
+        <div className="header--action full-centered">
+          <button onClick={() => setHideHelp(!hideHelp)} title={t('help')}>
+            <QuestionMarkCircleIcon className="icon-sm" />
+          </button>
```

Comment:
This is easy to overlook. I think we should show the help per default, and make it dismissible with a close button. Ideally, we could even save if the user has closed it, and then no longer show it per default.

Reply:
As dicsussed between us, I updated the code with e1429b7 to display the help on load when there are no match rules yet.

This shows the help when using the feature the first time, without us having to track any state explicitly.